compiler/rustc_passes/src/stability.rs

```diff
@@ -164,8 +167,24 @@ impl<'a, 'tcx> Annotator<'a, 'tcx> {
 
         let (stab, const_stab) = attr::find_stability(&self.tcx.sess, attrs, item_sp);
 
-        let const_stab = const_stab.map(|(const_stab, _)| {
+        let const_stab = const_stab.map(|(const_stab, const_span)| {
             let const_stab = self.tcx.intern_const_stability(const_stab);
+            match item_kind {
+                Some(ItemKind::Fn(ref fn_sig, _, _)) => {
+                    if !(fn_sig.header.constness == Constness::Const
+                        && fn_sig.header.abi != Abi::RustIntrinsic
+                        && fn_sig.header.abi != Abi::PlatformIntrinsic)
+                    {
+                        self.tcx
+                            .sess
+                            .struct_span_err(fn_sig.span, "Attributes `#[rustc_const_unstable]` and `#[rustc_const_stable]` require the function to be marked `const`")
+                            .span_label(const_span, "Const stability attribute specified here")
```

Comment:
Errors in the compiler start with a lowercase letter
```suggestion
                            .struct_span_err(fn_sig.span, "attributes `#[rustc_const_unstable]` and `#[rustc_const_stable]` require the function to be marked `const`")
                            .span_label(const_span, "const stability attribute specified here")
```